client/src/views/sidebar-menu/SidebarMenuPage.vue

```diff
@@ -299,6 +312,11 @@ async function createWorkspace(): Promise<void> {
 
 async function loadAll(): Promise<void> {
   const infoResult = await parsecGetClientInfo();
+  const savedFavorites = await storageManager.retrieveComponentData<WorkspacesPageSavedFavorites>(WORKSPACES_PAGE_DATA_KEY);
+
+  if (savedFavorites && savedFavorites.favoriteList !== undefined) {
```

Comment:
If we really can't store an array of strings, the serialization/deserialization should be done by the storageManager. Something like `JSON.stringify()`/`JSON.parse()` on all data that gets through `retrieveComponentData` or `saveComponentData`.
But we shouldn't have to remember to do this everytime we want to get the liste of favourites.